oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/session/SessionImpl.java

```diff
@@ -845,6 +846,12 @@ public UserManager getUserManager() throws RepositoryException {
         return sessionContext.getUserManager();
     }
 
+    @Override
+    @NotNull
+    public Set<Principal> getPrincipals() throws RepositoryException {
+        return sd.getAuthInfo().getPrincipals();
+    }
```

Comment:
We may want to check for consistency with other APIs. Maybe Iterable would be better.

Reply:
I already shared my view in https://issues.apache.org/jira/browse/OAK-11498?focusedCommentId=17928122&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17928122.